sway_libs/src/nft/extensions/meta_data/meta_data.sw

```diff
@@ -0,0 +1,68 @@
+library meta_data;
+
+dep meta_data_structures;
+
+use meta_data_structures::NFTMetaData;
+use ::nft::{errors::InputError, nft_core::NFTCore, nft_storage::{META_DATA, TOKENS}};
+use std::{hash::sha256, storage::{get, store}};
+
+pub trait MetaData<T> {
+    /// Returns the metadata for this token
+    #[storage(read)]
+    fn meta_data(self) -> Option<T>;
+    /// Creates new metadata for this token.
+    ///
+    /// # Arguments
+    ///
+    /// * `metadata` - The metadata value to be included in the new metadata.
+    #[storage(write)]
+    fn set_meta_data(self, metadata: Option<T>);
+}
+
+impl<T> MetaData<T> for NFTCore {
```

Comment:
How come this is implemented on the core instead of the meta data struct?

Reply:
All traits should be implemented for the NFTCore and not any external structs. The example matadata in this library isn't particularly useful and each project will likely have its own metadata. 

This way, devs can just call `my_token.set_meta_data(my_struct)` with any struct they create themselves without needing to implement their own metadata get and set functions. The trait for the core takes care of this.